### PR TITLE
feat: Add legal fragment

### LINKED ISSUE #190: Add Legal Section to Settings
Working on it :+1:

### PR DESCRIPTION
**Pull request by dreadpool2**
Fixes #190 

screenshot_20180703-164001

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        activity?.supportActionBar?.title = "Legal"
+        setHasOptionsMenu(true)
+    }
+
+    override fun onPreferenceTreeClick(preference: Preference?): Boolean {
+        if (preference?.key == resources.getString(R.string.key_privacy)) {
+            //Opens our app in play store
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Wrong comment

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            context?.let {
+                Utils.openUrl(it, PRIVACY_LINK)
+            }
+            return true
+        }
+        if (preference?.key == resources.getString(R.string.key_terms_of_service)) {
+            //Links to suggestion form
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: wrong comment

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            context?.let {
+                Utils.openUrl(it, TERMS_OF_SERVICE_LINK)
+            }
+            return true
+        }
+        if (preference?.key == resources.getString(R.string.key_cookie_policy)) {
+            //Logout Dialog shown
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: wrong comment

**Comment by dreadpool2** [comment]
PR Updated :+1:  Removed those comments

**Comment by dreadpool2** [comment]
@nikit19 @simarsingh24 @iamareebjamal Please review

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 4]
```diff
@@ -96,7 +96,7 @@ class ProfileFragment : Fragment() {
                 val bundle = Bundle()
                 bundle.putString(EMAIL, emailSettings)
                 fragment.arguments = bundle
-                activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.frameContainer, fragment)?.addToBackStack(null)?.commit()
+                activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.rootLayout, fragment)?.addToBackStack(null)?.commit()
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: why is this changed?

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Settings Fragment now covers the whole screen

File: app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt [hunk 5]
```diff
@@ -63,6 +72,11 @@ class SettingsFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
             showDialog()
             return true
         }
+        if (preference?.key == resources.getString(R.string.key_legal)) {
+            //Initialise Legal Fragment
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: this comment is not necessary (doesnt add any meaning) please remove it

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+        context?.let { ContextCompat.getColor(it, android.R.color.white) }?.let { view?.setBackgroundColor(it) }
+        return view
+    }
+
+    override fun onCreatePreferencesFix(savedInstanceState: Bundle?, rootKey: String?) {
+        // Load the preferences from an XML resource
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: while you are at it, can you remove this comment as well.
Comments should be for stuff that isn't obvious

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Sure will remove all comments :+1:

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Updated PR

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onDestroyView() {
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        activity?.supportActionBar?.title = "Settings"
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: That's not the question, this approach isn't robust. Rather, set the title in onResume of Settings

LegalFragment should not even know the existence of SettingsFragment

[the remark this answers is not in the document]
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Okay! Seems reasonable to do that :+1:

[the remark this answers is not in the document]
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: PR Updated

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        activity?.supportActionBar?.title = "Legal"
+        super.onResume()
+    }
+
+    override fun onDestroy() {
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        setHasOptionsMenu(false)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Same

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu?) {
+        menu?.setGroupVisible(R.id.profile_menu, false)
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onResume() {
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Same

File: app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt [hunk 10]
```diff
@@ -106,4 +116,19 @@ class SettingsFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
...
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        activity?.supportActionBar?.title = "Settings"
+        super.onResume()
+    }
+
+    override fun onDestroy() {
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(false)
+        activity?.supportActionBar?.title = "Profile"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Same problem as previously

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Yeah, this a problem actually. onResume method is not working in Profile Fragment.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Check why onResume isn't called

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Actually onResume works as it should. Whenever the app is on hold and again opened, the title is updated to Profile, but when we click the back button of SettingsFragment, it doesn't work peculiarly

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: onResume should still be called

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: I created a new project and tried, it is working flawlessly

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Was the same code utilized like are you just setting the title in the onResume method or anything else is being done ?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: I'm setting the title in onResume

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Backpress is also working? Like in my case the only problem was that on backpress of Settings Fragment , The profile fragment title would change to Settings and remain as that!

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: I will try again then :+1:

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: I pushed the code, onBackPress of Settings fragment it doesn't work still.

**Comment by dreadpool2** [comment]
@iamareebjamal 
Current status:-
videotogif_2018 07 09_13 17 29

**Comment by dreadpool2** [comment]
@iamareebjamal I used findFragmentByTag and overriden onBackPressed method. It works fine.

**Comment by dreadpool2** [comment]
@iamareebjamal PR Updated. onResume method is working :+1:

File: app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt [hunk 11]
```diff
@@ -29,13 +32,18 @@ class SettingsFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
         preferenceChange(evt)
     }
 
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+        view?.setBackgroundColor(Color.WHITE)
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: why not in xml?

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: This cant be done in settings.xml file actually

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: why?

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu?) {
+        menu?.setGroupVisible(R.id.profile_menu, false)
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onResume() {
+        val activity =  activity as? AppCompatActivity
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: why is this both in onResume and oncReate?

File: app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt (hunk 11 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: That xml file contains a PreferenceScreen, not a layout. Preference screen doesnt have any xml method to set the background. We do it programitically :+1:

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt (hunk 12 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: Resolved

File: app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.prefs.PreferenceChangeEvent
+import java.util.prefs.PreferenceChangeListener
+
+class LegalFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
+    val PRIVACY_LINK: String = "https://eventyay.com/privacy-policy/"
+    val TERMS_OF_SERVICE_LINK: String = "https://eventyay.com/terms/"
+    val COOKIE_POLICY_LINK: String = "https://eventyay.com/cookie-policy/"
+
+    override fun preferenceChange(evt: PreferenceChangeEvent?) {
+        preferenceChange(evt)
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/settings/LegalFragment.kt]: `evt` is not a meaningful name, can you please change it to something meaningful

File: app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt [hunk 14]
```diff
@@ -29,15 +32,16 @@ class SettingsFragment : PreferenceFragmentCompat(), PreferenceChangeListener {
...
     override fun onCreatePreferencesFix(savedInstanceState: Bundle?, rootKey: String?) {
         // Load the preferences from an XML resource
         setPreferencesFromResource(R.xml.settings, rootKey)
 
-        val activity = activity as? AppCompatActivity
-        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
-        activity?.supportActionBar?.title = "Settings"
-        setHasOptionsMenu(true)
-
         //Set Email
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: please remove comments which are very obvious

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/settings/SettingsFragment.kt]: Updated

**Comment by dreadpool2** [comment]
@iamareebjamal Please review
